core/input/input.cpp

```diff
@@ -1859,13 +1868,7 @@ void Input::set_fallback_mapping(const String &p_guid) {
 
 //platforms that use the remapping system can override and call to these ones
 bool Input::is_joy_known(int p_device) {
-	if (joy_names.has(p_device)) {
-		int mapping = joy_names[p_device].mapping;
-		if (mapping != -1 && mapping != fallback_mapping) {
-			return true;
-		}
-	}
-	return false;
+	return joy_names.has(p_device) && joy_names[p_device].is_known;
```

Comment:
```suggestion
	Joypad *joy = joy_names.getptr(p_device);
	return joy && joy->is_known;
```
For performance to avoid double lookup

Reply:
This isn't a hot path, so I wouldn't bother doing such micro-optimization which IMO hurts readability.